### PR TITLE
LUX 4.0.28

### PR DESCRIPTION
**Pull request by jon-kirwan**
## What / why
Update to 4.0.28.

## Visual Changes
None.

### CODE REVIEW

File: app/assets/javascripts/govuk_publishing_components/vendor/lux/lux-reporter.js [hunk 1]
```diff
@@ -1962,7 +1963,13 @@
       ].join("");
     }
     function getCustomerId() {
-      return String(_thisCustomerId) || "";
```
[the remark this answers is not in the document]
Comment (andysellick) [reply app/assets/javascripts/govuk_publishing_components/vendor/lux/lux-reporter.js:1965]: And thanks for picking this up @jon-kirwan, much appreciated.

[the remark this answers is not in the document]
Comment (jon-kirwan) [reply app/assets/javascripts/govuk_publishing_components/vendor/lux/lux-reporter.js]: Thanks for checking - I've changed it back to how it should be now

**Comment by jon-kirwan** [comment]
I pasted the output into the SpeedCurve debugging tool and received the following, which seems to be error free
```
     0 ms: Navigation started at 15/11/2024 17:32:32
 5,229 ms: lux.js v4.0.28 is initialising. Hover to view configuration.
 5,230 ms: POST beacon initialised.
 5,230 ms: Sample rate is 1%. This session is not being sampled.
 5,230 ms: Onload handler was triggered after 5230 ms.
 5,230 ms: LUX.addData("http-protocol", "http/1.1")
 5,230 ms: lux.js has finished initialising.
 5,230 ms: Received paint entry at 2383 ms
 5,230 ms: Received paint entry at 2383 ms
 5,231 ms: Received LCP entry at 2460 ms
 5,231 ms: Received LCP entry at 2485 ms
 5,231 ms: Received layout shift at 2377 ms with value of 0.127
 5,232 ms: Received layout shift at 2441 ms with value of 0.000
 5,232 ms: Received layout shift at 2447 ms with value of 0.001
 5,232 ms: Received layout shift at 5146 ms with value of 0.059
 5,232 ms: Received FID entry with duration of 32 ms
 6,092 ms: Received layout shift at 6089 ms with value of 0.061
 7,341 ms: Received layout shift at 7340 ms with value of 0.782
 7,426 ms: Received layout shift at 7421 ms with value of 0.290
24,621 ms: Unload handler was triggered.
24,621 ms: This beacon was not sent because the page visibility was hidden.
24,621 ms: POST beacon send() called.
32,788 ms: Unload handler was triggered.
32,788 ms: This beacon was not sent because the page visibility was hidden.
32,788 ms: POST beacon send() called.
```

File: app/assets/javascripts/govuk_publishing_components/vendor/lux/lux-reporter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (andysellick) [reply app/assets/javascripts/govuk_publishing_components/vendor/lux/lux-reporter.js]: Sorry @jon-kirwan think I used confusing language there - what I meant to say was that this function should remain as it previously was on the `main` branch, and any change introduced by this new version should be ignored.

[the remark this answers is not in the document]
Comment (jon-kirwan) [reply app/assets/javascripts/govuk_publishing_components/vendor/lux/lux-reporter.js]: No worries @andysellick, I've reverted the `getCustomerId` function to how it is on `main` e.g.
```
function getCustomerId() {
  return String(_thisCustomerId) || "";
}
```

[the remark this answers is not in the document]
Comment (andysellick) [reply app/assets/javascripts/govuk_publishing_components/vendor/lux/lux-reporter.js]: I'm not seeing anything in the diff, have you pushed?

**Comment by andysellick** [comment]
@jon-kirwan sorry I was being dense. Can you add a CHANGELOG entry as well? Thanks

File: app/assets/javascripts/govuk_publishing_components/vendor/lux/lux-reporter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (andysellick) [reply app/assets/javascripts/govuk_publishing_components/vendor/lux/lux-reporter.js]: Ignore me, I'm being silly.

**Pull request merged**